neqo-common/src/codec.rs

```diff
@@ -92,33 +82,43 @@ impl<'a> Decoder<'a> {
         Some(res)
     }
 
-    /// Decodes an unsigned integer of length 1..=8.
-    ///
-    /// # Panics
-    ///
-    /// This panics if `n` is not in the range `1..=8`.
-    pub fn decode_uint(&mut self, n: usize) -> Option<u64> {
-        assert!(n > 0 && n <= 8);
+    #[inline]
+    pub(crate) fn decode_n(&mut self, n: usize) -> Option<u64> {
```

Comment:
Would there be any further performance benefit in splitting this out into seven functions, e.g., 
```suggestion
    pub(crate) fn decode_1(&mut self, n: usize) -> Option<u64> {...}
    ...
    pub(crate) fn decode_8(&mut self, n: usize) -> Option<u64> {...}
```
each with optimized code for that length? (Is there a way to do this with a generic parameter more concisely?)